### ISSUE TITLE
lib/zocial/zocial.css is not included in the .jar file

### ISSUE THREAD

**Issue by AnhTuan-AiT** [keycloakify/keycloakify#418]
Hi, I tried building the starter project on Ubuntu

<b>Environment:</b>
```
Node: v16.14.0
Ubuntu: 20.04.6 LTS
```

When I built with Keycloak 22.0.1 and Keycloakify 8.0.0-rc.2, I get this error:
```
node:fs:1392
  handleErrorFromBinding(ctx);
  ^

Error: ENOENT: no such file or directory, scandir 'keycloakify-starter/build_keycloak/src/main/resources/theme/keycloakify-starter/tmp_suLeKsxId_202f15f6/keycloak/account/resources'
    at Object.readdirSync (node:fs:1392:3)
    at crawlRec (keycloakify-starter/node_modules/keycloakify/bin/tools/crawl.js:43:35)
    at crawl (keycloakify-starter/node_modules/keycloakify/bin/tools/crawl.js:65:5)
    at transformCodebase (keycloakify-starter/node_modules/keycloakify/bin/tools/transformCodebase.js:52:50)
    at keycloakify-starter/node_modules/keycloakify/bin/keycloakify/generateTheme/downloadKeycloakStaticResources.js:86:63
    at step (keycloakify-starter/node_modules/keycloakify/bin/keycloakify/generateTheme/downloadKeycloakStaticResources.js:56:23)
    at Object.next (keycloakify-starter/node_modules/keycloakify/bin/keycloakify/generateTheme/downloadKeycloakStaticResources.js:37:53)
    at fulfilled (keycloakify-starter/node_modules/keycloakify/bin/keycloakify/generateTheme/downloadKeycloakStaticResources.js:28:58) {
  errno: -2,
  syscall: 'scandir',
  code: 'ENOENT',
  path: 'keycloakify-starter/build_keycloak/src/main/resources/theme/keycloakify-starter/tmp_suLeKsxId_202f15f6/keycloak/account/resources'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

So I change Keycloak version to 21.1.2 and build successfully, but the .jar file misses this file <b>build_keycloak/src/main/resources/theme/keycloakify-starter/login/resources/resources_common
lib/zocial/zocial.css</b>, lead to the login form cannot display.

Finnally, I try Keycloak 22.0.1 and Keycloakify 9.0.0-rc.0. This time I see the file <b>build_keycloak/src/main/resources/theme/keycloakify-starter/login/resources/resources_common
lib/zocial/zocial.css</b> but the .jar file build fail because of this error:
```
node:child_process:828
    err = new Error(msg);
          ^

Error: Command failed: mvn package
    at checkExecSyncError (node:child_process:828:11)
    at Object.execSync (node:child_process:902:15)
    at keycloakify-starter/node_modules/keycloakify/bin/keycloakify/keycloakify.js:194:39
    at step (keycloakify-starter/node_modules/keycloakify/bin/keycloakify/keycloakify.js:56:23)
    at Object.next (keycloakify-starter/node_modules/keycloakify/bin/keycloakify/keycloakify.js:37:53)
    at fulfilled (keycloakify-starter/node_modules/keycloakify/bin/keycloakify/keycloakify.js:28:58) {
  status: 1,
  signal: null,
  output: [
    null,
    Buffer(715) [Uint8Array] [
       91,  27,  91,  49,  59,  51,  49, 109,  69,  82,  82,  79,
       82,  27,  91, 109,  93,  32,  69, 114, 114, 111, 114,  32,
      101, 120, 101,  99, 117, 116, 105, 110, 103,  32,  77,  97,
      118, 101, 110,  46,  10,  91,  27,  91,  49,  59,  51,  49,
      109,  69,  82,  82,  79,  82,  27,  91, 109,  93,  32, 106,
       97, 118,  97,  46, 108,  97, 110, 103,  46,  73, 108, 108,
      101, 103,  97, 108,  83, 116,  97, 116, 101,  69, 120,  99,
      101, 112, 116, 105, 111, 110,  58,  32,  85, 110,  97,  98,
      108, 101,  32, 116,
      ... 615 more items
    ],
    Buffer(0) [Uint8Array] []
  ],
  pid: 30770,
  stdout: Buffer(715) [Uint8Array] [
     91,  27,  91,  49,  59,  51,  49, 109,  69,  82,  82,  79,
     82,  27,  91, 109,  93,  32,  69, 114, 114, 111, 114,  32,
    101, 120, 101,  99, 117, 116, 105, 110, 103,  32,  77,  97,
    118, 101, 110,  46,  10,  91,  27,  91,  49,  59,  51,  49,
    109,  69,  82,  82,  79,  82,  27,  91, 109,  93,  32, 106,
     97, 118,  97,  46, 108,  97, 110, 103,  46,  73, 108, 108,
    101, 103,  97, 108,  83, 116,  97, 116, 101,  69, 120,  99,
    101, 112, 116, 105, 111, 110,  58,  32,  85, 110,  97,  98,
    108, 101,  32, 116,
    ... 615 more items
  ],
  stderr: Buffer(0) [Uint8Array] []
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Here is the <b>package.json</b> I use in 3 cases, only change version of Keycloak and Keycloakify:
```
{
  "name": "keycloakify-starter",
  "homepage": "https://starter.keycloakify.dev",
  "version": "4.7.0",
  "description": "A starter/demo project for keycloakify",
  "repository": {
    "type": "git",
    "url": "git://github.com/codegouvfr/keycloakify-starter.git"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "build-keycloak-theme": "yarn build && keycloakify",
    "storybook": "start-storybook -p 6006"
  },
  "keycloakify": {
    "extraThemeProperties": [
      "foo=bar"
    ],
    "extraThemeNames": [
      "keycloakify-starter-variant-1"
    ],
    "keycloakVersionDefaultAssets": "22.0.1"
  },
  "author": "u/garronej",
  "license": "MIT",
  "keywords": [],
  "dependencies": {
    "evt": "^2.4.15",
    "jwt-decode": "^3.1.2",
    "keycloak-js": "22.0.1",
    "keycloakify": "9.0.0-rc.0",
    "powerhooks": "^0.26.8",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "tsafe": "^1.6.0"
  },
  "devDependencies": {
    "@storybook/addon-actions": "^6.5.16",
    "@storybook/addon-essentials": "^6.5.16",
    "@storybook/addon-interactions": "^6.5.16",
    "@storybook/addon-links": "^6.5.16",
    "@storybook/builder-webpack5": "^6.5.16",
    "@storybook/manager-webpack5": "^6.5.16",
    "@storybook/node-logger": "^6.5.16",
    "@storybook/preset-create-react-app": "^4.1.2",
    "@storybook/react": "^6.5.16",
    "@storybook/testing-library": "^0.0.13",
    "@types/node": "^15.3.1",
    "@types/react": "18.0.9",
    "@types/react-dom": "18.0.4",
    "react-scripts": "5.0.1",
    "typescript": "~4.7.0"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ],
    "rules": {
      "react-hooks/exhaustive-deps": "off",
      "@typescript-eslint/no-redeclare": "off",
      "no-labels": "off"
    },
    "overrides": [
      {
        "files": [
          "**/*.stories.*"
        ],
        "rules": {
          "import/no-anonymous-default-export": "off"
        }
      }
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

Please help me resolve this issue~ Many thanks <3

**Comment by garronej** [comment]
Hello @AnhTuan-AiT,

Thank you for reaching out.

Could you elaborate on your reasons for wanting to change `keycloakVersionDefaultAssets`? Contrary to what you might assume, this value doesn't need to match the version of Keycloak you're using. The default Keycloak assets simply provide a fallback styling for pages you haven't customized. When I initially translated the default Keycloak theme to React, the Keycloak version was 11.0.3; hence the default version remains the same. However, I am planning to upgrade it to version 22 soon.

If you specifically want to use the assets from Keycloak 22.0.1, for instance to use the latest PatternFly version, follow these steps:

- [x] Update `"keycloakVersionDefaultAssets": "22.0.1"` in your `package.json`.
- [ ] Update your [`src/keycloak-theme/login/Template.tsx`](https://github.com/keycloakify/keycloakify-starter/blob/92b20fe74154ef8cf037f4b156eb3b2e5264a074/src/keycloak-theme/login/Template.tsx#L34-L44) file to actually use the default assets from [Keycloak 22.0.1](https://github.com/keycloak/keycloak/blob/d0fed02e1d59dfc98c8ecce4b4ab2332705f47ca/themes/src/main/resources/theme/keycloak/login/theme.properties#L4-L5). 

```tsx
    const { isReady } = usePrepareTemplate({
        "doFetchDefaultThemeResources": doUseDefaultCss,
        "styles": [
            `${url.resourcesCommonPath}/web_modules/@patternfly/react-core/dist/styles/base.css`,
            `${url.resourcesCommonPath}/web_modules/@patternfly/react-core/dist/styles/app.css`,
            `${url.resourcesCommonPath}/node_modules/patternfly/dist/css/patternfly.min.css`,
            `${url.resourcesCommonPath}/node_modules/patternfly/dist/css/patternfly-additions.min.css`,
            `${url.resourcesCommonPath}/node_modules/patternfly/dist/css/lib/pficon/pficon.css`,
            `${url.resourcesPath}/css/login.css`
        ],
        "htmlClassName": getClassName("kcHtmlClass"),
        "bodyClassName": undefined
    });
```

Further steps are theoretically optional but recommended for complete customization:

- [ ] Modify [`useGetClassName`](https://github.com/keycloakify/keycloakify/blob/main/src/login/lib/useGetClassName.ts) to match the property names in Keycloak 22.0.1.
- [ ] Eject all pages.
- [ ] Resolve type errors.
- [ ] Confirm that updating to the new asset version doesn't break your layout.

In summary, updating the assets is ideally something I should handle at the project level. I suggest focusing on building your theme using your own assets.

As for Keycloakify 9, it is not yet buildable and requires [this PR to be merged into the Keycloak project](https://github.com/keycloak/keycloak/pull/22317).

Hope this helps.

**Comment by AnhTuan-AiT** [comment]
That's right @garronej , I see that 11.0.3's login.css is a bit different so I want to try changing it to match the Keycloak version I'm using. I will wait for your update to version 22. 

Thank you very much for your support <3
